code/campaignclassic/src/main/java/com/adobe/marketing/mobile/ManualCarouselTemplateNotificationBuilder.java

```diff
@@ -0,0 +1,531 @@
+            newIndices =
+                    CampaignPushUtils.calculateNewIndices(
+                            centerImageIndex,
+                            imageUrls.size(),
+                            CampaignPushConstants.IntentActions.MANUAL_CAROUSEL_LEFT_CLICKED);
+        } else {
+            newIndices =
+                    CampaignPushUtils.calculateNewIndices(
+                            centerImageIndex,
+                            imageUrls.size(),
+                            CampaignPushConstants.IntentActions.MANUAL_CAROUSEL_RIGHT_CLICKED);
+        }
+
+        int newCenterIndex;
+        if (newIndices == null) {
+            Log.trace(
+                    CampaignPushConstants.LOG_TAG,
+                    SELF_TAG,
+                    "Unable to calculate new left, center, and right indices. Using default center"
+                            + " image index of 1.");
+            newCenterIndex = CampaignPushConstants.DefaultValues.CENTER_INDEX;
+        } else {
+            newCenterIndex = newIndices.get(1);
+        }
+
+        final ArrayList<CarouselPushTemplate.CarouselItem> items = new ArrayList<>();
+        final CarouselPushTemplate.CarouselItem centerCarouselItem =
+                new CarouselPushTemplate.CarouselItem(
+                        imageUrls.get(newCenterIndex),
+                        imageCaptions.get(newCenterIndex),
+                        imageClickActions.get(newCenterIndex));
+        items.add(centerCarouselItem);
+
+        final String messageId =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.MESSAGE_ID);
+        final String deliveryId =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.DELIVERY_ID);
+        final int badgeCount = intentExtras.getInt(CampaignPushConstants.IntentKeys.BADGE_COUNT);
+        final int visibility = intentExtras.getInt(CampaignPushConstants.IntentKeys.VISIBILITY);
+        final int importance = intentExtras.getInt(CampaignPushConstants.IntentKeys.IMPORTANCE);
+        final String channelId =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.CHANNEL_ID);
+        final String notificationBackgroundColor =
+                intentExtras.getString(
+                        CampaignPushConstants.IntentKeys.NOTIFICATION_BACKGROUND_COLOR);
+        final String titleTextColor =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.TITLE_TEXT_COLOR);
+        final String expandedBodyTextColor =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.EXPANDED_BODY_TEXT_COLOR);
+        final String smallIcon =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.SMALL_ICON);
+        final String smallIconColor =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.SMALL_ICON_COLOR);
+        final String customSound =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.CUSTOM_SOUND);
```

Comment:
What are your thoughts on creating two different methods `createNotificationBuilderFromIntent(Context, Intent...)` and `createNotificationBuilderFromTemplate(Context, CarouselPushTemplate.....)`  and then moving this logic for fetching the parameters from tempate and intent there? I feel that will reduce the number of parameters that you need to pass around (especially given java lacking named parameter support).

Reply:
i am open to doing this. what if i do it in a future pr? reason being that i would need to make this change for each template class which would touch quite a few of the files.

Reply:
Sure. I am aligned on doing this after we wrap up the remind later part.